### PR TITLE
sorter(ticdc): optimize sorter and fix corner case of updating resolvedTs

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #10457 

### What is changed and how it works?
1. optimize the handle events logic in sorter engine.
2. fix updating resolved ts when some span is missing.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

**Comment by hongyunyan** [comment]
@hicqu

File: cdc/processor/sourcemanager/sorter/pebble/event_sorter.go [hunk 1]
```diff
@@ -371,17 +369,76 @@ type tableState struct {
...
+// It will commit the batch when the size of the batch is larger than batchCommitSize or
+// the time since the last commit is larger than batchCommitInterval.
+// It will also update the resolved ts of the table when the batch is committed.
+// Considering commit is a heavy operation, we make [fetch and decode event] and
+// [do commit and update resolved ts] as two separate goroutines to make pipeline.
+func (s *EventSorter) handleEvents(
+	id int, db *pebble.DB, inputCh <-chan eventWithTableID,
+) {
+	batchCh := make(chan *DBBatchEvent, 1000)
+	go s.batchCommitAndUpdateResolvedTs(batchCh, id)
```
Comment (asddongmen) [inline cdc/processor/sourcemanager/sorter/pebble/event_sorter.go:437]: It is better use `s.wg.Add(1)` here to manager this goroutine.

**Review by asddongmen — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Comment by flowbehappy** [comment]
Please create another issue instead of using ref.

**Pull request merged**